addons/html_editor/static/src/others/embedded_components/plugins/toggle_plugin/toggle_plugin.js

```diff
@@ -0,0 +1,627 @@
+import { getEmbeddedProps } from "@html_editor/others/embedded_component_utils";
+import { Plugin } from "@html_editor/plugin";
+import { baseContainerGlobalSelector } from "@html_editor/utils/base_container";
+import { closestBlock } from "@html_editor/utils/blocks";
+import { removeClass } from "@html_editor/utils/dom";
+import { isEmptyBlock, isParagraphRelatedElement } from "@html_editor/utils/dom_info";
+import {
+    childNodes,
+    children,
+    closestElement,
+    firstLeaf,
+    lastLeaf,
+    selectElements,
+} from "@html_editor/utils/dom_traversal";
+import { parseHTML } from "@html_editor/utils/html";
+import { childNodeIndex, nodeSize } from "@html_editor/utils/position";
+import { withSequence } from "@html_editor/utils/resource";
+import { _t } from "@web/core/l10n/translation";
+import { renderToString } from "@web/core/utils/render";
+import { uuid } from "@web/views/utils";
+
+const toggleSelector = "[data-embedded='toggle']";
+const titleSelector = "[data-embedded-editable='title']";
+const contentSelector = "[data-embedded-editable='content']";
+
+export class TogglePlugin extends Plugin {
+    static id = "toggle_list";
+    static dependencies = [
+        "baseContainer",
+        "delete",
+        "dom",
+        "embeddedComponents", // toggle is an embedded component.
+        "hint", // needed to clean custom toggle hints.
+        "history",
+        "selection",
+        "split",
+    ];
+    resources = {
+        hints: [
+            withSequence(1, {
+                selector: `${toggleSelector} ${titleSelector} > *`,
+                text: _t("Toggle title"),
+            }),
+        ],
+        move_node_selectors_blacklist: `${toggleSelector} ${titleSelector} *`,
+        powerbox_items: [
+            {
+                commandId: "insertToggleList",
+                categoryId: "structure",
+            },
+        ],
+        shortcuts: [{ hotkey: "control+enter", commandId: "controlEnter" }],
+        user_commands: [
+            {
+                id: "insertToggleList",
+                title: _t("Toggle List"),
+                description: _t("Hide Text under foldable toggles"),
+                icon: "fa-caret-square-o-right",
+                isAvailable: (node) => !closestElement(node, `${toggleSelector} ${titleSelector}`),
+                run: () => {
+                    this.insertToggleList();
+                },
+            },
+            {
+                id: "controlEnter",
```

Comment:
It's weird that the name of command is its shortcut. Can't you find a better name for this? Conceptually, a command is something you could trigger from a command palette or through `vim` or something. How would you name that feature/action if you were to call it in `vim`? You'd certainly not call it "controlEnter".

Reply:
I took inspiration from `tab_plugin` and I thought the idea was to use the hotkey as an id to avoid conflicting commands (so if I want to use `controlEnter`, I have a conflicting id error and I know to take care not to overlap with the other feature, then I rename it `controlEnter[something]`)

but I'll rename it then :)

Reply:
Haha, I see, yes, the tabulation plugin also uses "wrong" command names indeed... ^^' I guess it should ideally use something like tabulate/untabulate... oops ^^'